src/components/KEmptyState/KEmptyState.vue

```diff
@@ -1,143 +1,173 @@
+        class="empty-state-message"
       >
-        <slot name="cta">
-          <KButton
-            appearance="primary"
-            size="small"
-            @click.prevent="() => handleClick && handleClick()"
-          >
-            {{ ctaText }}
-          </KButton>
+        <slot name="default">
+          <p>
+            {{ message }}
+          </p>
         </slot>
       </div>
     </div>
-  </section>
+    <div
+      v-if="actionButtonVisible && (actionButtonText || $slots.action)"
+      class="empty-state-action"
+    >
+      <slot name="action">
+        <KButton
+          :disabled="actionButtonDisabled"
+          @click="$emit('action-click')"
+        >
+          {{ actionButtonText }}
+        </KButton>
+      </slot>
+    </div>
+  </div>
 </template>
 
 <script lang="ts" setup>
+import { computed, type PropType } from 'vue'
+import { AnalyticsIcon, WarningIcon, CogIcon, FileEmptyIcon, SearchIcon, KongIcon } from '@kong/icons'
+import { KUI_COLOR_TEXT_NEUTRAL, KUI_COLOR_TEXT_WARNING, KUI_ICON_SIZE_60 } from '@kong/design-tokens'
 import KButton from '@/components/KButton/KButton.vue'
-import KIcon from '@/components/KIcon/KIcon.vue'
-import { KUI_COLOR_TEXT_NEUTRAL_STRONGEST, KUI_ICON_SIZE_80 } from '@kong/design-tokens'
+import { EmptyStateIconVariants } from '@/types'
+import type { EmptyStateIconVariant } from '@/types'
 
-defineProps({
-  isError: {
-    type: Boolean,
-    default: false,
-  },
-  iconSize: {
+type EmptyStateIcon = typeof AnalyticsIcon // all icons are the same type so we can use any of them
+
+const props = defineProps({
+  title: {
     type: String,
-    default: KUI_ICON_SIZE_80,
+    default: '',
   },
-  icon: {
+  message: {
     type: String,
     default: '',
   },
-  ctaIsHidden: {
+  actionButtonVisible: {
     type: Boolean,
-    default: false,
+    default: true,
   },
-  ctaText: {
+  actionButtonText: {
     type: String,
     default: '',
   },
-  handleClick: {
-    type: Function,
-    default: null,
-  },
-  iconColor: {
-    type: String,
-    default: '',
+  actionButtonDisabled: {
+    type: Boolean,
+    default: false,
   },
-  iconSecondaryColor: {
-    type: String,
-    default: '',
+  iconVariant: {
+    type: String as PropType<EmptyStateIconVariant>,
+    default: EmptyStateIconVariants.Default,
   },
 })
-</script>
 
-<style lang="scss" scoped>
+defineEmits(['action-click'])
+
+const getEmptyStateIcon = computed((): EmptyStateIcon => {
+  switch (props.iconVariant) {
+    case EmptyStateIconVariants.Default:
+      return AnalyticsIcon
+    case EmptyStateIconVariants.Error:
+      return WarningIcon
+    case EmptyStateIconVariants.Config:
+      return CogIcon
+    case EmptyStateIconVariants.File:
+      return FileEmptyIcon
+    case EmptyStateIconVariants.Search:
+      return SearchIcon
+    case EmptyStateIconVariants.Kong:
+      return KongIcon
+    default:
+      return AnalyticsIcon // default variant in case of invalid value
```

Comment:
Importing all of these icons is increasing the bundle size even when most of them won't be used in the component instance.

Reply:
Removed a few, only left 3 - the rest can be slotted